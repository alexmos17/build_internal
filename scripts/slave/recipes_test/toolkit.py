# Copyright (c) 2013 The Chromium Authors. All rights reserved.
# Use of this source code is governed by a BSD-style license that can be
# found in the LICENSE file.

"""Test inputs for recipes/toolkit.py"""

def basic_test(api):
  return {
    'build_properties': api.tryserver_build_properties(
      repository="https://github.com/toolkitchen/toolkit"
    ),
    'test_data': {
      'gen step(gen_steps.py)': (0, [
        {'name': 'bogus', 'cmd': ['bogus.py', '--arg']}
      ])
    }
  }
