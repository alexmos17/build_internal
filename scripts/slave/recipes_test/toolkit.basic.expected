[
  {
   "cmd": ["[BUILD_ROOT]/scripts/slave/git_setup.py", "--path", "[SLAVE_BUILD_ROOT]/toolkit", "--url", "https://github.com/toolkitchen/toolkit.git"],
   "name": "git setup",
   "seed_steps": ["git setup", "git fetch", "git update-ref", "git clean", "git checkout", "git submodule"]
  },
  {
   "cmd": ["git", "--work-tree", "[SLAVE_BUILD_ROOT]/toolkit", "--git-dir", "[SLAVE_BUILD_ROOT]/toolkit/.git", "fetch", "origin", "--recurse-submodules"],
   "name": "git fetch",
   "seed_steps": []
  },
  {
   "cmd": ["git", "--work-tree", "[SLAVE_BUILD_ROOT]/toolkit", "--git-dir", "[SLAVE_BUILD_ROOT]/toolkit/.git", "update-ref", "refs/heads/master", "origin/master"],
   "name": "git update-ref",
   "seed_steps": []
  },
  {
   "cmd": ["git", "--work-tree", "[SLAVE_BUILD_ROOT]/toolkit", "--git-dir", "[SLAVE_BUILD_ROOT]/toolkit/.git", "clean", "-f", "-d", "-X"],
   "name": "git clean",
   "seed_steps": []
  },
  {
   "cmd": ["git", "--work-tree", "[SLAVE_BUILD_ROOT]/toolkit", "--git-dir", "[SLAVE_BUILD_ROOT]/toolkit/.git", "checkout", "-f", "master"],
   "name": "git checkout",
   "seed_steps": []
  },
  {
   "cmd": ["git", "--work-tree", "[SLAVE_BUILD_ROOT]/toolkit", "--git-dir", "[SLAVE_BUILD_ROOT]/toolkit/.git", "submodule", "update", "--init", "--recursive"],
   "name": "git submodule",
   "seed_steps": []
  },
  {
   "cmd": ["[SLAVE_BUILD_ROOT]/toolkit/buildbot/gen_steps.py", "--factory-properties", "{\"recipe\": \"toolkit\"}", "--build-properties", "{\"blamelist\": [\"cool_dev1337@chromium.org\"], \"patchset\": 1, \"rietveld\": \"https://chromiumcodereview.appspot.com\", \"issue\": 12853011, \"repository\": \"https://github.com/toolkitchen/toolkit\"}", "--output-json", "/path/to/tmp/json"],
   "cwd": "[SLAVE_BUILD_ROOT]/toolkit",
   "name": "gen step(gen_steps.py)"
  },
  {
   "cmd": ["bogus.py", "--arg"],
   "name": "bogus",
   "seed_steps": ["bogus"]
  }
]